app/models/server_puppet_data.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class ServerPuppetData
+  attr_reader :client
+
+  def initialize
+    @client = PuppetdbClient.new(Settings.PUPPETDB_ENDPOINT)
+  end
+
+  def self.run
+    new.load_data
+  end
+
+  def load_data
+    Server.find_each do |server|
```

Comment:
I believe `.all` loads every object into memory, while `find_each` loads batches of a certain number, so `find_each` would be more efficient for large numbers of things returned.